src/utils/utils.ts

```diff
@@ -62,53 +64,177 @@ function standardErrorReplacer(key: string, value: any) {
+  const maxColumnLengths: Record<string, number> = {};
+
+  const optionColumns =
+    options?.columns != null
+      ? Array.isArray(options.columns)
+        ? options.columns
+        : Object.keys(options.columns)
+      : undefined;
+
+  // Initialize maxColumnLengths with header lengths if headers are provided
+  if (optionColumns != null) {
+    for (const column of optionColumns) {
+      maxColumnLengths[column] = Math.max(
+        options?.columns?.[column] ?? 0,
+        column.length,
+      );
+    }
+  }
+
+  // Precompute max column lengths by iterating over the rows first
+  for (const row of rows) {
+    for (const column in options?.columns ?? row) {
+      if (row[column] != null) {
+        row[column] = encodeNonPrintable(row[column].toString());
+      }
+      // Row[key] is definitely a string or null after this point due to encodeNonPrintable
+      const cellValue: string | null = row[column];
+      // Null or '' will both cause cellLength to be 3
+      const cellLength =
+        cellValue == null || cellValue.length === 0 ? 3 : cellValue.length; // 3 is length of 'N/A'
+      maxColumnLengths[column] = Math.max(
+        maxColumnLengths[column] || 0,
+        cellLength, // Use the length of the encoded value
+      );
+    }
+  }
+
+  // After this point, maxColumnLengths will have been filled with all the necessary keys.
+  // Thus, the column keys can be derived from it.
+  const columns = Object.keys(maxColumnLengths);
+  // If headers are provided, add them to your output first
+  if (optionColumns != null) {
+    for (let i = 0; i < optionColumns.length; i++) {
+      const column = optionColumns[i];
+      const maxColumnLength = maxColumnLengths[column];
+      // Options.headers is definitely defined as optionHeaders != null
+      if (!Array.isArray(options!.columns)) {
+        options!.columns![column] = maxColumnLength;
+      }
+      if (includeHeaders) {
+        output += column.padEnd(maxColumnLength);
+        if (i !== optionColumns.length - 1) {
+          output += '\t';
+        } else {
+          output += '\n';
+        }
+      }
+    }
+  }
+
+  for (const row of rows) {
+    let formattedRow = '';
+    if (options.includeRowCount) {
+      formattedRow += `${++rowCount}\t`;
+    }
+    for (const column of columns) {
+      // Assume row[key] has been already encoded as a string or null
+      const cellValue =
+        row[column] == null || row[column].length === 0 ? 'N/A' : row[column];
+      formattedRow += `${cellValue.padEnd(maxColumnLengths[column] || 0)}\t`;
+    }
+    output += formattedRow.trimEnd() + '\n';
+  }
+
+  return output;
+}
+
+/**
+ * Formats a message suitable for output.
+ * @param msg - The msg that needs to be formatted.
+ * @see {@link outputTableFormatter} for information regarding usage where `msg.type === 'table'`.
+ * @returns
+ */
 function outputFormatter(msg: OutputObject): string | Uint8Array {
```

Comment:
Question, is `Uint8Array` used for `raw`?

Reply:
yes, for `CommandGet` of `secrets` domain